java/core/src/main/java/sleeper/core/partition/PartitionsBuilder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.partition;
+
+import sleeper.core.range.Range;
+import sleeper.core.range.Region;
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.PrimitiveType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A convenience class for specifying partitions.
+ * <p>
+ * Note that a shorthand is used for cases where we have a schema with only one row key field.
+ * This will not be useful in the general case.
+ */
+public class PartitionsBuilder {
+
+    private final Schema schema;
+    private final List<PrimitiveType> rowKeyTypes;
+    private final List<Partition> partitions = new ArrayList<>();
+
+    public PartitionsBuilder(Schema schema) {
+        this.schema = schema;
+        rowKeyTypes = schema.getRowKeyTypes();
+    }
+
+    public Partition partition(String id, Object min, Object max) {
```

Comment:
I worry that it's not clear enough that this whole class should only be used for schemas with one dimensional row keys? There's a risk that someone will try to use it with min being a List of the minimums for row key field 1 and row key field 2.

Reply:
That gets validated in RangeFactory right? I think that's enough now this is in src/test?